bin/kafka-run-class.sh

```diff
@@ -22,6 +22,12 @@ fi
 
 base_dir=$(dirname $0)/..
 
+# run kafka-env.sh
+KAFKA_ENV=$base_dir/config/kafka-env.sh
```

Comment:
`kafka-run-class.sh` is used by many scripts. Are there any concerns about using a single set of configs here? For example, you might want zookeeper, Kafka, console producer/consumer, etc. to all run with different heap settings, which means you can't set those parameters in this shared file. Would sourcing these make more sense one level up, which allows you to separate out `config/kafka-env.sh` from `config/zookeeper-env.sh`, as well as any others that make sense to separate out? The obvious drawback is having to remember to add that to every single script.

Also, any concern about when this gets included? For example, there are some tests for whether environment variables are already set in scripts like `kafka-server-start.sh`, which then set defaults if they are not yet set. Since this env script is not sourced until `kafka-run-class.sh` is executed, these variables will already be set and the env script can't be used to set custom defaults. The option described above would also address this problem since this env script could be sourced before anything else is done.

I'm not convinced anything needs to change in the patch, I just want to make sure we've thought through the implications of the current approach.